`editor::SelectPrevious` at end of buffer crashes

### Summary

<!-- Please insert a one line summary of the issue below -->

<!-- Include all steps necessary to reproduce from a clean Zed installation. Be verbose -->
### Steps to trigger the problem:
1. Open a new buffer
2. Type the character `a`
3. DO NOT add a new line after it
4. Trigger `editor::SelectPrevious` twice.

###  Actual Behavior:

Crashes.

### Expected Behavior:

Shouldn't crash.

### Zed Version and System Specs

can't reproduce in stable
reproduced in preview and nightly:

Zed: v0.173.1 (Zed Preview)
Zed: v0.174.0 (Zed Nightly b13498a5dd279a2eeb92a42b0ad996e85094e0c0)
OS: Linux X11 arch unknown
Memory: 60.3 GiB
Architecture: x86_64
GPU: AMD Radeon Graphics (RADV RAPHAEL_MENDOCINO) || radv || Mesa 24.3.4-arch1.1

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

<details><summary>Zed.log</summary>

<!-- Click below this line and paste or drag-and-drop your log-->
```
2025-02-05T23:44:55.805697607-03:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-02-05T23:48:11.007788926-03:00 [ERROR] no such worktree
2025-02-05T23:48:18.123585904-03:00 [ERROR] {
  "thread": "main",
  "payload": "called `Option::unwrap()` on a `None` value",
  "location_data": {
    "file": "crates/multi_buffer/src/multi_buffer.rs",
    "line": 7087
  },
  "backtrace": [
    "zed::reliability::init_panic_hook::{{closure}}::h8232216154ba4bab+118352913",
    "std::panicking::rust_panic_with_hook::hfe205f6954b2c97b+92343912",
    "std::panicking::begin_panic_handler::{{closure}}::h6cb44b3a50f28c44+92343075",
    "std::sys::backtrace::__rust_end_short_backtrace::hf1c1f2a92799bb0e+92335497",
    "rust_begin_unwind+92342244",
    "core::panicking::panic_fmt::h3d8fc78294164da7+7298099",
    "core::panicking::panic::hec978767ec2d35ff+7298236",
    "core::option::unwrap_failed::hba6b08832f9ce30b+7297945",
    "<multi_buffer::ReversedMultiBufferChunks as core::iter::traits::iterator::Iterator>::next::hbfca4b9fc5a7e73a+66619321",
    "multi_buffer::MultiBufferSnapshot::reversed_bytes_in_range::h965f3c07de738e92+34427805",
    "editor::Editor::select_previous::he47841a7124b4e57+39479132",
    "editor::Editor::select_previous::he47841a7124b4e57+39478824",
    "editor::element::register_action::{{closure}}::h6f0223ff4ef855bb+38018605",
    "gpui::window::Window::dispatch_action_on_node::hf8b37a7f8ff09711+53904688",
    "core::ops::function::FnOnce::call_once{{vtable.shim}}::h4c03c976faef318e+50295707",
    "gpui::app::App::flush_effects::h1d923d7a85591518+52850288",
    "gpui::window::Window::new::{{closure}}::h94c26ad02bb9012c+53832889",
    "gpui::platform::linux::x11::window::X11WindowStatePtr::handle_input::h8861fb9650cd91f4+53612742",
    "gpui::platform::linux::x11::client::X11Client::handle_event::h7f7d5a30741723f7+53552985",
    "gpui::platform::linux::x11::client::X11Client::process_x11_events::hdc74688921c879c9+53527244",
    "<core::cell::RefCell<calloop::sources::DispatcherInner<S,F>> as calloop::sources::EventDispatcher<Data>>::process_events::hf76554a185a7999a+49565019",
    "<gpui::platform::linux::x11::client::X11Client as gpui::platform::linux::platform::LinuxClient>::run::ha23045fef59b6b74+53589206",
    "gpui::platform::linux::platform::<impl gpui::platform::Platform for P>::run::h83ddc1f3f7706f4e+53203997",
    "gpui::app::Application::run::hcc6adca8671c366a+114882021",
    "zed::main::h3c6ebbc78475349b+118406097",
    "std::sys::backtrace::__rust_begin_short_backtrace::h2598106130cf963f+117496755",
    "std::rt::lang_start::{{closure}}::h343f697b26056175+116556329",
    "std::rt::lang_start_internal::h5e7c81cecd7f0954+92268645",
    "main+118423260",
    "__libc_start_main+35193592575308",
    "_start+10873646"
  ],
  "app_version": "0.174.0",
  "release_channel": "nightly",
  "target": "x86_64-unknown-linux-gnu",
  "os_name": "Linux X11",
  "os_version": "arch unknown",
  "architecture": "x86_64",
  "panicked_on": 1738810098123,
  "system_id": "5f06c84b-cb3d-4c89-88a6-31b82da45990",
  "installation_id": "fadcdfa9-3c07-4366-9c19-b759b99bdf6e",
  "session_id": "9fa0b525-8f60-422e-b500-01239454faa5"
}

```
<!-- Click above this line and paste or drag-and-drop your log--></details>

--- a/crates/editor/src/editor_tests.rs
+++ b/crates/editor/src/editor_tests.rs
@@ -5363,6 +5363,21 @@ async fn test_select_previous_with_single_caret(cx: &mut gpui::TestAppContext) {
 }
 
 #[gpui::test]
+async fn test_select_previous_empty_buffer(cx: &mut gpui::TestAppContext) {
+    init_test(cx, |_| {});
+
+    let mut cx = EditorTestContext::new(cx).await;
+    cx.set_state("aˇ");
+
+    cx.update_editor(|e, window, cx| e.select_previous(&SelectPrevious::default(), window, cx))
+        .unwrap();
+    cx.assert_editor_state("«aˇ»");
+    cx.update_editor(|e, window, cx| e.select_previous(&SelectPrevious::default(), window, cx))
+        .unwrap();
+    cx.assert_editor_state("«aˇ»");
+}
+
+#[gpui::test]
 async fn test_select_previous_with_multiple_carets(cx: &mut gpui::TestAppContext) {
     init_test(cx, |_| {});
 

--- a/crates/multi_buffer/src/multi_buffer.rs
+++ b/crates/multi_buffer/src/multi_buffer.rs
@@ -7121,7 +7121,7 @@ impl<'a> Iterator for ReversedMultiBufferChunks<'a> {
             self.offset -= 1;
             Some("\n")
         } else {
-            let chunk = self.current_chunks.as_mut().unwrap().next().unwrap();
+            let chunk = self.current_chunks.as_mut().unwrap().next()?;
             self.offset -= chunk.len();
             Some(chunk)
         }

